### PR TITLE
Added tests for some wallet send events

### LINKED ISSUE #20411: Cover wallet send events with tests
# Problem

Events in `status_im/context/wallet/send/events.cljs` aren't covered with tests. 
Since this is the most crucial part of wallet, it should be properly tested.

## Implementation

These events should be covered:

- [x] :wallet/clean-send-data (Milad)
- [x]       :wallet/select-address-tab (Milad)
- [x]       :wallet/suggested-routes-success (Milad)
- [x]      :wallet/suggested-routes-error (Milad)
- [x]      :wallet/clean-suggested-routes (Milad)
- [x]      :wallet/clean-send-address (Milad)
- [x]      :wallet/clean-send-amount (Milad)
- [x]      :wallet/clean-disabled-from-networks (Milad)
- [x]      :wallet/clean-from-locked-amounts (Milad)
- [x]      :wallet/select-send-address (Milad)
- [x]      :wallet/update-receiver-networks (Volodymyr)
- [x]      :wallet/set-token-to-send (Volodymyr)
- [x]      :wallet/edit-token-to-send (Volodymyr)
- [x]      :wallet/clean-selected-token (Milad)
- [x]      :wallet/clean-selected-collectible (Milad)
- [x]      :wallet/set-collectible-to-send (Volodymyr)
- [x]      :wallet/set-collectible-amount-to-send (Volodymyr)
- [x]      :wallet/set-token-amount-to-send (Volodymyr)
- [x]      :wallet/disable-from-networks (Milad)
- [x]      :wallet/lock-from-amount (Milad)
- [x]      :wallet/unlock-from-amount (Milad)
- [x]      :wallet/reset-network-amounts-to-zero (Milad)
- [ ]      :wallet/get-suggested-routes (Milad)
- [x]      :wallet/add-authorized-transaction (Milad)
- [ ]      :wallet/close-transaction-progress-page (Milad)
- [ ]      :wallet/send-transaction (Milad)
- [x]      :wallet/select-from-account (Milad)

## Acceptance Criteria

Tests written for all the important events

### PR DESCRIPTION
**Pull request by vkjr**
Partially fixes #20411

### Summary

Tests added for events:
```
 :wallet/update-receiver-networks 
 :wallet/set-token-to-send 
 :wallet/edit-token-to-send 
 :wallet/set-collectible-to-send 
 :wallet/set-collectible-amount-to-send 
 :wallet/set-token-amount-to-send 
```

status: ready

### CODE REVIEW

File: src/status_im/contexts/wallet/send/events.cljs
Comment (vkjr) [inline src/status_im/contexts/wallet/send/events.cljs:215]: The only change here is that `:token-display-name` is now set only if `token` is non-nil. Before it was set always but was `nil`

**Comment by vkjr** [comment]
Manual QA can be skipped because there is no change in functionality

**Comment by mmilad75** [comment]
This PR should be merged after https://github.com/status-im/status-mobile/pull/20535 is merged, because there is an issue with `match?` which is going to be fixed on #20535

File: src/status_im/contexts/wallet/send/events.cljs
Comment (ulisesmac) [reply src/status_im/contexts/wallet/send/events.cljs:215]: Please just make sure the send flow is not broken if we start it from any possible entry point

**Review by ulisesmac — APPROVED** [review]
@vkjr Thanks for covering these events with tests!

**Review by J-Son89 — APPROVED** [review]
Great work!

File: src/status_im/contexts/wallet/send/events.cljs
Comment (vkjr) [reply src/status_im/contexts/wallet/send/events.cljs:215]: There are 2 possible entry points to initiate this event - when the user selects a token from the wallet screen or when selects an asset after pressing the "send" button within a specific account. I checked both, they work without changes.

**Comment by mmilad75** [comment]
@vkjr There is still one issue with `match?` which @ilmotta is checking at the moment.

`(h/deftest-event :wallet/suggested-routes-error
  [event-id dispatch]
  (is (match? {:key :val
               :key2 :val2} 
              {:key :val
               :key2 :val2
               :key3 :val3})))`
               
This test is not failing which it should. So please wait until Icaro fixes it and rebase your branch

**Comment by vkjr** [comment]
@mmilad75, sure, I will wait until fix is merged and will check my tests again. Thanks!

**Comment by ilmotta** [comment]
@vkjr, @mmilad75,

Per investigation, `match?` uses a default matcher called `embeds` in map instances. This matcher ignores extra keys. Maybe this is the default behavior because it's what Rich believes (open maps are to be preferred) and the library is part of Nubank, so who knows ;)

In the newest version of `3.9.0` (we are in `3.8.8`) we have the `nested-equals` matcher in cases where we want to be super strict. Assertion would look like this:

```clojure
(is (match? (matchers/nested-equals expected) actual))
```

Therefore, upgrading the lib should be safe and then you guys can use that new nested matcher. We talked about changing the default map matcher to be more strict, but I don't know if it's possible to change the defaults in the lib. If not, we can always register our own custom cljs.test directive like `match?`, although painful, it's good to know we have this option.

**Comment by vkjr** [comment]
@ilmotta, thanks for the details! 
In general, I'm okay with the default behavior of `match?` because it checks that the key is present and this is a functionality tests need the most. For the cases when I had to check if the key was removed I used `get-in`.

**Pull request merged**
